app/controllers/service_provider_controller.rb

```diff
@@ -0,0 +1,13 @@
+class ServiceProviderController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  def update
+    if FeatureManagement.use_dashboard_service_providers?
+      ServiceProviderUpdater.new.run
+      SecureHeadersWhitelister.new.run
```

Comment:
Whenever we change the list of service providers, we must also update the CSP header so that the new service providers are whitelisted on forms. So every time we run ServiceProviderUpdater during runtime, we must also execute SecureHeadersWhitelister.

We don't need the override in this controller. We just need to alter the in-memory CSP configuration for all other controllers that *do* need the override.

> Would it make more sense to move the SERVICE_PROVIDERS assignment out of its own initializer and into the SecureHeadersWhitelister class, such that it sets the constant to the proper value based on whether or not FeatureManagement.use_dashboard_service_providers? is true?

I don't understand that suggestion.